pycozmo/conn.py

```diff
@@ -76,26 +80,32 @@ def run(self) -> None:
             self.handle_frame(frame)
 
     def handle_frame(self, frame: Frame) -> None:
+        frame_trusted = True
         for pkt in frame.pkts:
-            self.handle_pkt(pkt)
+            frame_trusted = self.handle_pkt(pkt)
+            if not frame_trusted:
+                return
```

Comment:
No need to return here. The stream of packets is independent from the acknowledgement, which is for the stream of packets in the other direction

```
          ------- packets ------>
          <-- acknowledgments ---
engine                              robot
          <------ packets -------
          --- acknowledgments -->
```

Reply:
You are right, sorry, I think I meant to have that with one indent less, so skip the acknowledgement if the package is repeated.

I think I will also include the logic to manage the acknowledgement in the send window. I only avoided it because it means to include a new variable to track the progress of the acknowledgement (and I felt lazy about it I guess haha). This is necessary because the robot sends an accumulate acknowledgement (I think it's called like that), which does not match the address/id of the packets received, but rather counts how many have been received in order.